man/kbmadm.1m

```diff
@@ -0,0 +1,249 @@
+.Sh SYNOPSIS
+.Nm
+.Cm create-zpool
+.Op Fl g Ar guid
+.Op Fl t Ar template
+--
+.Ar zpool-create-args
+.Nm
+.Cm recover
+.Op Fl c Ar cfgnum
+.Ar dataset
+.Nm
+.Cm unlock
+.Op Fl r
+.Ar dataset
+.Nm
+.Cm recovery add
+.Op Fl f
+.Op Fl r Ar recovery_token
+.Fl t Ar template
+.Ar dataset
+.Nm
+.Cm recovery list
+.Op Fl p
+.Nm
+.Cm recovery activate
+.Ar dataset
+.Nm
+.Cm recovery cancel
+.Ar dataset
+.Sh DESCRIPTION
+.Nm
+is used to to manage
+.Xr kbmd 1M .
+.Sh SUBCOMMANDS
+.Bl -tag -width ""
+.\"
+.\" create-zpool
+.\"
+.It Xo
+.Nm
+.Cm create-zpool
+.Op Fl g Ar guid
+.Op Fl t Ar template
+.Ar dataset
+--
+.Ar zpool-create-args
+.Xc
+Create a new zpool with the root dataset encrypted.
+.Pp
+If
+.Ar guid
+is not specified, an uninitialized PIV token must be present on the system.
+.Pp
+The
+.Nm
+.Cm zpool-create
+subcommand will initialize the PIV token, call the
+.Sy register-pivtoken
+plugin method, and then create the zpool.
+.Bl -tag -width Fl
+.It Fl g Ar guid
+Do not initialize a PIV token.
+Instead, use the PIV token identified by
+.Ar guid
+to create the ebox.
+The PIV token must be inserted on the system at the time of the
+command invocation or it will fail.
+.It Fl t Ar template
+Read the recovery template from
+.Ar template .
+When initializing a PIV token, the
+.Sy register-pivtoken
+plugin method normally specifies the recovery configuration to use.
+Specifying this option will override any recovery configuration provided by the
+.Sy register-pivtoken
+plugin method.
+.Pp
+If the
+.Fl t Ar template
+option is not specified and either the
+.Fl g Ar guid
+option is also not specified, or the
+.Sy register-pivtoken
+plugin method does not specify a recovery configuration, the
+resulting ebox will not contain a recovery configuration.
+.It Ar zpool-create-args
+Arguments to pass through to the
+.Nm zpool
+.Cm create
+command.
+Arguments/options must be separated from the
+.Nm
+.Cm create-zpool
+options by two dashes
+.Pq Dq --
+\(em even if no
+.Nm
+.Cm zpool-create
+arguments are given.
+Any option listed in the
+.Nm zpool
+.Cm create
+subcommand in
+.Xr zpool
+may be passed along.
+.Pp
+Note that stdin of the
+.Nm zpool
+.Cm create
+command is not available.
+.El
+.\"
+.\" recover
+.\"
+.It Xo
+.Nm
+.Cm recover
+.Op Fl c Ar cfgnum
+.Ar dataset
+.Xc
+Start a recovery session for
+.Ar dataset
+using the active recovery configuration template.
+If the active reconfiguration template contains multiple recovery
+configuartions, the
+.Fl c Ar cfgnum
+option may be used to specifify which configuration to use.
+If the
+.Fl c Ar cfgnum
+option is not present, the recovery process will prompt the operator to
+first select a recovery configuration from the active recovery template
+before starting the challenge/response process.
+.\"
+.\" unlock
+.\"
+.It Xo
+.Nm
+.Cm unlock
+.Op Fl r
+.Ar dataset
+.Xc
+Unlock the ebox associated with the given dataset, load the key from the ebox,
+and mount the dataset (if appropriate).
```

Comment:
Currently, it only mounts if the dataset is a top-level dataset.  In my earlier comments, I suggested that we should allow unlocking without mounting.  That is, this should follow `zfs load-key` not `zfs mount -la`.

Reply:
What if we added a flag (-m ?) to do the 'if a top-level dataset, mount after unlocking' -- as I said in earlier comments, my concern here is that I've not been able to convince myself that doing a `zfs mount -a` after importing and unlocking all the pools will have the same effect as mounting (effectively) after importing each one (and might result in breaking some configurations) -- which is why I wanted to keep the process as close as possible to what happens today.

Reply:
We can defer implementation of this.  If a real-world case appears, we can add a "do not mount" option.

Reply:
confirmed